Library/Homebrew/cmd/update.rb

```diff
@@ -275,12 +276,15 @@ def select_formula key
       else
         path.basename(".rb").to_s
       end
-    end.sort
+    end.sort.uniq
   end
 
   def dump_formula_report key, title
     formula = select_formula(key)
+    formula.map! {|f| Formula.factory(f) rescue nil }
+    formula.select! {|f| f && f.installed? }
```

Comment:
Should this be conditional on `key == :M` (or do you want to stop showing new packages?)